space/internal/components/aclnotifications/notifications.go

```diff
@@ -71,9 +75,13 @@ func (n *aclNotificationSender) Run(ctx context.Context) (err error) {
 }
 
 func (n *aclNotificationSender) Close(ctx context.Context) (err error) {
-	if err := n.batcher.Close(); err != nil {
-		logger.Errorf("failed to close batcher, %s", err)
-	}
+	go func() {
+		n.mx.Lock()
```

Comment:
batcher is thread safe

Reply:
It doesn't relate to badger, notification can be send in two different processes (when space `loader` is inited and in `UpdateAcl`) in acl object manager. Space loader can close components like acl notification service, while another process (`UpdateAcl`) is still executed. So we avoid racing here